runtime/dart_init.cc

```diff
@@ -336,7 +336,11 @@ Dart_Isolate IsolateCreateCallback(const char* script_uri,
     }
     // Entry script path (file:// is stripped).
     entry_path = std::string(script_uri + strlen(kFileUriPrefix));
-    if (!running_from_source) {
+    if (StringEndsWith(entry_path, ".dill")) {
```

Comment:
Can you clarify in comments how this works with kernel file named 'kernel_blob.bin'  as defined by kKernelAssetKey here?

Reply:
done